File: src/main/java/org/cobbzilla/util/graphics/ImageTransformConfig.java
package org.cobbzilla.util.graphics;

import static org.cobbzilla.util.daemon.ZillaRuntime.die;

public class ImageTransformConfig {

    private int height;
    private int width;

    public ImageTransformConfig(String config) {
        final int xpos = config.indexOf('x');
        try {
            width = Integer.parseInt(config.substring(xpos + 1));
            height = Integer.parseInt(config.substring(0, xpos));
        } catch (Exception e) {
            die("invalid config (expected WxH): " + config + ": " + e, e);
        }
    }

    public int getHeight() {
        return this.height;
    }

    public int getWidth() {
        return this.width;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public void setWidth(int width) {
        this.width = width;
    }
}

File: src/main/java/org/cobbzilla/util/graphics/ImageTransformConfig.kt
package org.cobbzilla.util.graphics

import org.cobbzilla.util.daemon.ZillaRuntime.die

class ImageTransformConfig(config: String) {

    var height: Int = 0
    var width: Int = 0

    init {
        val xpos = config.indexOf('x')
        try {
            width = Integer.parseInt(config.substring(xpos + 1))
            height = Integer.parseInt(config.substring(0, xpos))
        } catch (e: Exception) {
            die<Any>("invalid config (expected WxH): $config: $e", e)
        }

    }
}
